gollm/factory.go

```diff
@@ -21,18 +21,21 @@ import (
 	"os"
 )
 
-// NewClient builds an Client based on the LLM_CLIENT env var
-func NewClient(ctx context.Context) (Client, error) {
-	s := os.Getenv("LLM_CLIENT")
-	if s == "" {
-		return nil, fmt.Errorf("LLM_CLIENT is not set")
-	}
-	u, err := url.Parse(s)
-	if err != nil {
-		return nil, fmt.Errorf("parsing LLM_CLIENT URL: %w", err)
+// NewClient builds an Client based on the LLM_CLIENT env var or the provided providerID. ProviderID (if not empty) overrides the provider from LLM_CLIENT env var.
+func NewClient(ctx context.Context, providerID string) (Client, error) {
```

Comment:
Cool - I imagine one day we will extend the LLM_CLIENT "uri" to be more than just a providerID, but an obvious extension here if/when we do!